pep-0723.rst

```diff
@@ -101,50 +94,61 @@ The following are some of the use cases that this PEP wishes to support:
 Specification
 =============
 
-Any Python script may assign a variable named ``__pyproject__`` to a multi-line
-*double-quoted* string literal (``"""``) containing a valid TOML document. The
-variable MUST start at the beginning of the line and the opening of the string
-MUST be on the same line as the assignment. The closing of the string MUST be
-on a line by itself, and MUST NOT be indented.
+This PEP defines a metadata comment block format inspired [2]_ loosely by
+`reStructuredText Directives`__.
 
-When there are multiple ``__pyproject__`` variables defined, tools MUST produce
-an error.
+__ https://docutils.sourceforge.io/docs/ref/rst/directives.html
 
-The TOML document MUST NOT contain multi-line double-quoted strings, as that
-would conflict with the Python string containing the document. Single-quoted
-multi-line TOML strings may be used instead.
+Any Python script may have a top-level comment block that starts with the line
+``# /// TYPE`` where ``TYPE`` determines how to process the content, and ends
+with the line ``# ///``. Every line between these two lines MUST be a comment
+starting with ``#``. If there are characters after the ``#`` then the first
+character MUST be a space. The embedded content is formed by taking away the
+first two characters of each line if the second character is a space, otherwise
+just the first character (which means the line consists of only a single
+``#``).
 
-This is the canonical regular expression that MUST be used to parse the
+Tools reading embedded metadata MAY respect the standard Python encoding
+declaration. If they choose not to do so, they MUST process the file as UTF-8.
+
+The first type of metadata block is named ``pyproject`` which represents a
+subset of content that one would see in a ``pyproject.toml`` file.
```

Comment:
I think "subset" is slightly misleading here, as the "[run]" table isn't valid in pyproject.toml, and as far as I can tell this PEP doesn't seek to make it legal -- so it is *similar* data, but not a strict subset.